### DISCUSSION TITLE
Tips on debugging a regex?

### DISCUSSION THREAD

**Discussion by hershkoy** [microsoft/Recognizers-Text#2701] [General]
I am in the process of adding support to a new language (Hebrew).
The current code is in this [draft pull request](https://github.com/microsoft/Recognizers-Text/pull/2699).

I currently have some tests that are failing. I found out that I can run specific tests using the ava match command, that is a huge time saver:
```
Recognizers-Text\JavaScript>ava --match='*0.08*'

  index-number » Number - Hebrew - NumberModel - "במשקל 0.08"
  Result.Resolution.value
  Difference:
  - '8'
  + '0.08'
```

So now that I know that a test fails, and that I can run it standalone, I could use some advice on what is the best way to debug.

1) How can I know which regex from the yaml file is used to extract the number from the sentence?
2) Is there a way to know the 'exploded' version of the nested regex from @1? Maybe using a breakpoint while running the test, but in which file should I do the breakpoint?

I checked the code of other languages in order to understand how they deal with 0.08. I think that it is using the `AllFloatRegex`:
```
English => '{AllIntRegex}(\s+point){AllPointRegex}'
Dutch => def: '{AllIntRegex}(\s+komma){AllPointRegex}'
```
But I don't understand how the '.' character is matched.

Any help is much appreciated
Thanks!

**Comment by tellarin** [comment]
One tip is adding identifiers momentarily to the regexes, so you can check wich regex matched or not.
Another useful strategy is to test similar utterances in English and see how the system deals with them and how that behaves differently from the new culture implementation.
Regarding breakpoints, you can add a "Debug": "true" property to a given spec and the tests will break in that case during debug, then you can follow execution step by step.
